planner/core/planbuilder.go

```diff
@@ -2467,6 +2467,24 @@ func (b *PlanBuilder) buildAnalyzeFullSamplingTask(
 		if colsInfo, ok := colsInfoMap[physicalID]; ok {
 			execColsInfo = colsInfo
 		}
+		filterSkipColumnTypes := func(origin []*model.ColumnInfo) (result []*model.ColumnInfo) {
+			skipTypes := b.ctx.GetSessionVars().AnalyzeSkipColumnTypes
```

Comment:
The background auto-analyze goroutine's `sctx` is initialized at the beginning, so can it read the latest variable value here?